smack-integration-test/src/main/java/org/jivesoftware/smackx/muc/MultiUserChatRolesAffiliationsPrivilegesIntegrationTest.java

```diff
@@ -794,23 +811,26 @@ public void mucTestDefaultRoleForAffiliationInMembersOnlyRoom() throws Exception
 
         createMembersOnlyMuc(mucAsSeenByOne, nicknameOne);
 
-        final SimpleResultSyncPoint adminResultSyncPoint = new SimpleResultSyncPoint();
+        final SimpleResultSyncPoint allOccupantsDetectedSyncPoint = new SimpleResultSyncPoint();
+        final Set<EntityFullJid> expectedOccupants = Set.of(jidOne, jidTwo, jidThree);
         mucAsSeenByOne.addParticipantStatusListener(new ParticipantStatusListener() {
             @Override
-            public void adminGranted(EntityFullJid participant) {
-                adminResultSyncPoint.signal();
+            public void joined(EntityFullJid participant) {
+                if (mucAsSeenByOne.getOccupants().containsAll(expectedOccupants)) {
+                    allOccupantsDetectedSyncPoint.signal();
+                }
             }
         });
 
         try {
             mucAsSeenByOne.grantMembership(conTwo.getUser().asBareJid());
             mucAsSeenByOne.grantMembership(conThree.getUser().asBareJid());
+            mucAsSeenByOne.grantAdmin(conTwo.getUser().asBareJid());
 
             mucAsSeenByTwo.join(nicknameTwo);
             mucAsSeenByThree.join(nicknameThree);
-            mucAsSeenByOne.grantAdmin(conTwo.getUser().asBareJid());
-            adminResultSyncPoint.waitForResult(timeout);
-            assertEquals(mucAsSeenByOne.getOccupantsCount(), 3, "Unexpected occupant count in room " + mucAddress);
+
+            assertResult(allOccupantsDetectedSyncPoint, "Expected " + conTwo.getUser() + " to observe all of these occupants in room " + mucAddress + ", but not all of them appear to be in: " + expectedOccupants.stream().map(Object::toString).collect(Collectors.joining(", ")));
```

Comment:
It's conOne that does the observing, no?

```suggestion
            assertResult(allOccupantsDetectedSyncPoint, "Expected " + conOne.getUser() + " to observe all of these occupants in room " + mucAddress + ", but not all of them appear to be in: " + expectedOccupants.stream().map(Object::toString).collect(Collectors.joining(", ")));
```

Reply:
Absolutely. I actually intended that, but `con<tab>` apparently added the wrong reference, without me noticing.